pycbc/waveform/waveform.py

```diff
@@ -775,6 +775,73 @@ def get_td_waveform_from_fd(rwrap=0.2, **params):
                                            params['f_lower']))
     return hp, hc
 
+def get_td_det_waveform_from_fd_det(template=None, rwrap=0.2, **params):
+    """ Return time domain version of fourier domain approximant which
+    includes detector response, with padding and tapering at the start
+    of the waveform.
+
+    Parameters
+    ----------
+    rwrap: float
+        Cyclic time shift parameter in seconds. A fudge factor to ensure
+        that the entire time series is contiguous in the array and not
+        wrapped around the end.
+    params: dict
+        The parameters defining the waveform to generator.
+        See `get_fd_det_waveform_sequence`.
+
+    Returns
+    -------
+    dict
+        The detector-frame waveform (with detector response) in time
+        domain. Keys are requested data channels.
+    """
+    kwds = props(template, **params)
+    # determine the duration to use
+    full_duration = duration = pnutils.get_imr_duration(
```

Comment:
We shouldn't hardcode this. What we need is to have a length estimator for the approximant and for it to be advertised using the plugin. @WuShichao Can you add the requisite function (maybe Imrphenomd used here is fine!) in the bbhx plugin and update it's setup.py so that it advertises a length estimator? That way you don't need to hardcode here. That may not be appropriate for other approximants.